src/main/java/com/rcv/Logger.java

```diff
@@ -44,145 +53,110 @@
   private static final String EXECUTION_LOG_FILE_NAME = "rcv_%g.log";
   // first value here is bytes per MB and the second is max MB for the log file
-  private static final Integer EXECUTION_LOG_FILE_MAX_SIZE_BYTES = 1000000 * 50;
+  private static final Integer EXECUTION_LOG_MAX_SIZE_BYTES = 1000000 * 50;
   // how many execution files to keep
   private static final Integer EXECUTION_LOG_FILE_COUNT = 2;
   // cache for the execution logger
   private static java.util.logging.Logger executionLogger;
   // cache for the tabulation logger
-  private static java.util.logging.Logger tabulationLogger;
+  private static java.util.logging.FileHandler tabulationHandler;
   // cache for the tabulation logger
-  private static java.util.logging.Logger guiLogger;
+  private static java.util.logging.Handler guiHandler;
 
   // function: setup
   // purpose: initialize logging module
   // throws: IOException if unable to open output log file
   static void setup() throws IOException {
     // create and cache default logger
     executionLogger = java.util.logging.Logger.getLogger(EXECUTION_LOGGER_NAME);
-    // remove any loggers the system may have installed
-    for (Handler handler : executionLogger.getHandlers()) {
-      executionLogger.removeHandler(handler);
+    executionLogger.setLevel(Level.FINE);
+
+    // set root console handler to use our formatter (just to keep things tidy and consistent)
+    java.util.logging.Logger rootLogger = executionLogger.getParent();
+    for(Handler handler : rootLogger.getHandlers()) {
+      handler.setFormatter(new LogFormatter());
     }
+
     // logPath is where default file logging is written
     // "user.dir" property is the current working directory, i.e. folder from whence the rcv jar
     // was launched
-    String logPath = Paths.get(System.getProperty("user.dir"), EXECUTION_LOG_FILE_NAME).toString();
-    // formatter specifies how logging output lines should appear
-    LogFormatter formatter = new LogFormatter();
+    Path logPath = Paths.get(System.getProperty("user.dir"),
+        EXECUTION_LOG_FILE_NAME).toAbsolutePath();
+
     // fileHandler writes formatted strings to file
     FileHandler fileHandler =
-        new FileHandler(logPath, EXECUTION_LOG_FILE_MAX_SIZE_BYTES, EXECUTION_LOG_FILE_COUNT, true);
-    fileHandler.setFormatter(formatter);
-    fileHandler.setLevel(Level.FINE);
-    // create a consoleHandler to writes formatted strings to console for debugging
-    ConsoleHandler consoleHandler = new ConsoleHandler();
-    consoleHandler.setFormatter(formatter);
-    consoleHandler.setLevel(Level.FINE);
-    // add the handlers
-    executionLogger.addHandler(consoleHandler);
+        new FileHandler(logPath.toString(),
+            EXECUTION_LOG_MAX_SIZE_BYTES,
+            EXECUTION_LOG_FILE_COUNT,
+            true);
+    // use our custom formatter
+    fileHandler.setFormatter(new LogFormatter());
```

Comment:
Might make more sense and be more memory-efficient to just instantiate a `static final` of the formatter and use it for all.

Reply:
dig